src/transformStateWithClones.js

```diff
@@ -7,7 +7,22 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let newState = { ...state };
+  const states = [];
+
+  for (const key of actions) {
```

Comment:
Consider renaming the variable `key` to something more descriptive, like `action`, to better reflect its purpose in the context of the loop.